packages/webapp/src/treasury/components/treasury-delegate-levels-info.tsx

```diff
@@ -1,5 +1,114 @@
+import {
+    Asset,
+    assetToString,
+    Text,
+    useMasterPool,
+    useMemberStats,
+    useTreasuryStats,
+} from "_app";
+
+enum RankLabel {
+    NumberedLevel = "N",
+    Chief = "Chief",
+    HeadChief = "Head-chief",
+}
+
 export const TreasuryDelegateLevelsInfo = () => {
-    return <div>{/* Delegate Levels Table goes Here! */}</div>;
+    const { data: memberStats } = useMemberStats();
+    const { data: pool } = useMasterPool();
+    const { data: treasuryStats } = useTreasuryStats();
+
+    if (
+        !memberStats ||
+        !pool ||
+        !treasuryStats ||
+        memberStats.ranks.length < 2
+    ) {
+        return null;
+    }
+
+    // we don't care about the initial round, it just indicates
+    // how many people participated in the election, the real
+    // delegates starts from the index 1
+    const electedRanks = memberStats.ranks.slice(1);
+    const electedRanksSize = electedRanks.length;
+
+    const totalDistributionAmount =
+        (treasuryStats.quantity * pool.monthly_distribution_pct) / 100;
+    const levelDistribution = totalDistributionAmount / electedRanksSize;
+
+    const calculateAndRenderRankLevelComponent = (
+        rankDelegatesCount: number,
+        index: number
+    ) => {
+        const currentRank = index + 1;
+
+        // we need to consider all the next rank delegates in the distribution
+        const nextRankDelegatesCount = electedRanks
+            .slice(currentRank)
+            .reduce((a, b) => a + b, 0);
+
+        const rankAmountQuantity =
+            levelDistribution / (rankDelegatesCount + nextRankDelegatesCount);
+        const rankAmount = { ...treasuryStats, quantity: rankAmountQuantity };
+
+        console.info("nextRankDelegatesArray", electedRanks.slice(currentRank));
+        console.info("currentRank", currentRank);
+        console.info("nextRankDelegatesCount", nextRankDelegatesCount);
+        console.info("rankAmountQuantity", rankAmountQuantity);
```

Comment:
question: do you want to keep these in here for some reason?

Reply:
nope! good catch!